shopelectro/tests/tests_models.py

```diff
@@ -39,27 +40,35 @@ def test_get_brands_content(self):
             self.assertEquals(tag.group, brand_group)
             self.assertIn(product, tag.products.all())
 
-    def test_filter_by_products_equality(self):
+    def test_filter_by_products(self):
+        sort_by_id = partial(sorted, key=lambda x: x.id)
         products = self.get_products()
         self.assertEquals(
-            set(Tag.objects.filter_by_products(products)),
-            set(chain.from_iterable(product.tags.all() for product in products)),
-        )
-
-    def test_filter_by_products_quantity(self):
-        products = self.get_products()
-        self.assertEquals(
-            Tag.objects.filter_by_products(products).count(),
-            len(set(chain.from_iterable(product.tags.all() for product in products))),
+            sort_by_id(Tag.objects.filter_by_products(products)),
+            sort_by_id(list(set(
+                chain.from_iterable(product.tags.all() for product in products)
+            ))),
         )
 
 
-class QueriesNumber(TransactionTestCase):
+class QueryQuantities(TransactionTestCase):
+    """Test quantity of db-queries for different methods."""
 
     fixtures = ['dump.json']
 
-    def test_tag_get_brands(self):
-        """Perform two query to fetch tags and related products."""
-        products = list(Product.objects.all())
+    def test_get_brands_from_tags(self):
+        """Perform two queries to fetch tags, its products."""
+        product = list(Product.objects.all())
         with self.assertNumQueries(2):
-            Tag.objects.get_brands(products)
+            Tag.objects.get_brands(product)
+
+    def test_get_brands_from_products(self):
+        """Perform a lot of queries, so we should fetch brands from the Tag model."""
+        products = Product.objects.all().prefetch_related('tags', 'tags__group')
```

Comment:
emm, but what about transaction.Atomic?

Reply:
@duker33 How is a transaction related to the number of queries? Transaction is about db consistency only

Reply:
@artemiy312 , y, it's true, sorry